src/allocator/memserver_allocator.cpp

```diff
@@ -342,12 +429,14 @@ void Memserver_Allocator::deallocate(uint64_t regionId, uint64_t offset) {
     message << "Error While deallocating dataitem : ";
     // call NVMM to destroy the data item
     Heap *heap = 0;
+    //printf("PNS: %s %d\n",__FUNCTION__,__LINE__);
 
     HeapMap::iterator it = get_heap(regionId, heap);
     if (it != heapMap->end()) {
         NVMM_PROFILE_START_OPS()
-        heap->Free(offset);
-        NVMM_PROFILE_END_OPS(Heap_Free)
+        EpochOp op(em);
```

Comment:
If the delayed_free_thread == 0 , we should should call normal heap free without epochOp